### ISSUE TITLE
Can't use SHA as `ref`

### ISSUE THREAD

**Issue by chrisluedtke** [convictional/trigger-workflow-and-wait#26]
README.md:
![image](https://user-images.githubusercontent.com/20371880/140581461-6d4b1e19-49b1-44f4-b1be-657cf1f886d1.png)

When passing a SHA as `ref`, I get:
```
{
  "message": "No ref found for: 64d9f66f1df87cd18c204770810acaec836dfadb",
  "documentation_url": "https://docs.github.com/rest/reference/actions#create-a-workflow-dispatch-event"
}
```

[Github docs](https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event):
> `ref` Required. The git reference for the workflow. The reference can be a branch or tag name.

**Comment by chrisluedtke** [comment]
Related: https://github.com/github/docs/issues/590

**Comment by lwille** [comment]
Indeed, we need to be able to pass the workflow's `branch` separately from the `ref` (if `ref` should be a SHA).

**Comment by ThePinzon** [comment]
Whats the status on this?  It would be fantastic to be able to use a sha.

**Comment by andrew-from-toronto** [comment]
Why can't we give a hash? Is this related to dispatched workflows not showing branch/commit hash on the workflow run list?

**Comment by benjamincitrin** [comment]
I believe this has been solved?
